EIPS/eip-epbs.md

```diff
@@ -0,0 +1,260 @@
+- `process_execution_payload` is now called as a separate helper when receiving a `SignedExecutionPayloadEnvelope` on the P2P layer. This function in particular checks that the HTR of the resulting beacon state coincides with the committed one in the payload envelope. 
+
+Forkchoice changes include:
+
+#### Constants
+
+| Name                 | Value       |
+| -------------------- | ----------- |
+| `PAYLOAD_TIMELY_THRESHOLD` | `PTC_SIZE/2` (=`uint64(256)`) | 
+| `INTERVALS_PER_SLOT` | `4` |
+| `PROPOSER_SCORE_BOOST` | `20` | 
+| `PAYLOAD_WITHHOLD_BOOST` | `40` | 
+| `PAYLOAD_REVEAL_BOOST` | `40` | 
+
+#### Containers
+
+```python
+class ChildNode(Container):
+    root: Root
+    slot: Slot
+    is_payload_present: bool
+```
+
+The class `LatestMessage` is modified to track the slot instead of an epoch:
+
+```python
+@dataclass(eq=True, frozen=True)
+class LatestMessage(object):
+    slot: Slot
+    root: Root
+```
+
+The class `Store` is modified to track the following fields
+```
+    execution_payload_states: Dict[Root, BeaconState] = field(default_factory=dict)
+    ptc_vote: Dict[Root, Vector[uint8, PTC_SIZE]] = field(default_factory=dict)
+    payload_withhold_boost_root: Root
+    payload_withhold_boost_full: bool
+    payload_reveal_boost_root: Root
+```
+
+- A new handler `on_payload_attestation_message` is called when receiving a `PayloadAttestationMessage` from the P2P network. 
+- A new handler `on_execution_payload` is called when receiving a `SignedExecutionPayloadEnvelope` from the P2P network. 
+
+### Engine API
+
+No changes needed
+
+## Rationale
+
+### Staked Builders
+
+Being a builder is a new attribution of validators. As such builders are staked in the beacon chain. This allows for in-protocol trustless enforcement of the builder's payment to the proposer. Alternatively, payment could be enforced in the Execution Layer at the cost of adding the corresponding EL consensus changing logic. Payments in the EL have the advantage of not requiring the builder to periodically submit deposit transactions to replenish their validator balance. Both systems require availability of funds before the payload is revealed: in the CL because the funds are staked, and in the EL because the payment much be done in the first transaction of the block. 
+
+### Delayed validation
+
+The Payload Timeliness Committee members do not need to validate the execution payload before attesting to it. They perform basic checks like verifying the builder's signature and the correct blockhash is included. This takes away the full execution payload validation from the hot path of validation of an Ethereum block, giving the next proposer 6 seconds (`SECONDS_PER_SLOT * 2 // INTERVALS_PER_SLOT`) to validate the payload and every other validator 9 seconds (`SECONDS_PER_SLOT * 3 // INTERVALS_PER_SLOT`). A tradeoff from the user experience point of view is that transactions have a longer delay into being confirmed on-chain as validly included.
```

Comment:
"basic checks such as"

Reply:
"verifying that the builder's signature and the correct blockhash are included"

Reply:
"A tradeoff from the user experience point of view is that transactions have a longer delay into being confirmed on-chain as validly included." bit confusing. Also I think it's possibly misleading to talk about "confirmation" without making explicit the confirmation rule you are referring to. Here you seem to say "the confirmation that some protocol-parties have checked validity", in which case the next proposer at `n+1` who builds on the payload at `n` would be the first party to give you this confirmation (but can you trust them?)

Reply:
> "verifying that the builder's signature and the correct blockhash are included"

Actually I think this is semantically different than what I say in the text. I added a comma instead. I am verifying the actual signature, not that it's included.

Reply:
> ...  Also I think it's possibly misleading to talk about "confirmation"...
Agreed, but the sentence is important, this is the first complain I get when describing the protocol and delayed validation. The fact that a transaction will take longer to be included in a block and longer to be in the head of the chain. I accept rephrasings... just don't know how to put it better in a single short sentence.

Reply:
> > ...  Also I think it's possibly misleading to talk about "confirmation"...
>
> Agreed, but the sentence is important, this is the first complain I get when describing the protocol and delayed validation. The fact that a transaction will take longer to be included in a block and longer to be in the head of the chain. I accept rephrasings... just don't know how to put it better in a single short sentence.

How about "From a user UX perspective, a transaction included in slot `n` by the builder is not widely validated until the proposer of slot `n+1` releases their beacon block on top of block `n` first, and the attesters of slot `n+1` vote on this beacon block as the head of the chain."